pkg/apis/core/v1beta1/helper/condition_builder.go

```diff
@@ -30,6 +31,9 @@ type ConditionBuilder interface {
 	WithReason(reason string) ConditionBuilder
 	WithMessage(message string) ConditionBuilder
 	WithCodes(codes ...gardencorev1beta1.ErrorCode) ConditionBuilder
+	WithClock(clock clock.Clock) ConditionBuilder
+	// Deprecated: Use WithClock(...) instead.
+	// TODO(oliver-goetz): Remove in a future release.
```

Comment:
Why don't we remove this `NowFunc` already now? I don't see much value in deprecating this now and removing it only later? Wouldn't make this things much easier/faster now?

Reply:
Sounds good 👍 
I just followed @plkokanov suggestion from the issue. In the end there are only 3 `NowFunc` usages left.

Reply:
Tbh if it only has 3 usages, then we can probably clean it up now. Sorry for not mentioning this sooner.

Reply:
No worries, it is removed now.